modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -424,10 +434,34 @@ private CompletableFuture<Void> updateLogicalTopology(CmgRaftService service) {
                 }));
     }
 
-    private void handleCancelInit(CancelInitMessage msg) {
+    private void handleCancelInit(CancelInitMessage msg, Consumer<Throwable> onError) {
         LOG.info("CMG initialization cancelled [reason={}]", msg.reason());
+        this.scheduledExecutor.execute(() -> this.destroyCmgWithEvents().exceptionally(toFunction(onError)));
```

Comment:
Semantically speaking, `exceptionally` is used to translate exceptions, not to consume them. `whenComplete` would be a better choice here.